src/lib/dialog/dialog.ts

```diff
@@ -52,6 +48,34 @@ export class MdDialog {
       @Optional() @SkipSelf() private _parentDialog: MdDialog) { }
 
   /**
+   * Opens a modal dialog containing the given TemplateRef.
+   * @param TemplateRef to load into the dialog
+   * @param config Extra configuration options.
+   * @returns Reference to the newly-opened dialog.
+   */
+  openFromTemplateRef<T>(templateRef: TemplateRef<T>, config?: MdDialogConfig): MdDialogRef<T> {
+    config = _applyConfigDefaults(config);
+
+    let overlayRef = this._createOverlay(config);
```

Comment:
Not sure I get the issue here. Isn't it down to doing a `ref instanceof TemplateRef` check and then taking the proper route under the hood?